lib/config.js

```diff
@@ -0,0 +1,204 @@
+// Copyright 2017 TODO Group. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+const Ajv = require('ajv')
+const fetch = require('node-fetch')
+const findConfig = require('find-config')
+const fs = require('fs')
+const jsonfile = require('jsonfile')
+const lodash = require('lodash')
+const path = require('path')
+const yaml = require('js-yaml')
+
+const Rules = require('../rules/rules')
+const RuleInfo = require('./ruleinfo')
+const Fixes = require('../fixes/fixes')
+
+function isAbsoluteURL(url) {
+  try {
+    const u = new URL(url)
+    if (u.host !== '') {
+      return true
+    }
+  } catch (e) {}
+  return false
+}
+
+/**
+ * Load a ruleset config from the specified location.
+ *
+ * @param {string} [configLocation] A URL, file location, or directory containing a repolinter config file
+ * @param {array} [processed] list of config files already processed, used to prevent loops
+ * @returns {Object} The loaded repolinter json config
+ * @throws Will throw an error if unable to parse config or if config is invalid
+ */
+async function loadConfig(configLocation, processed = []) {
+  if (configLocation == null) {
+    throw new Error('must specify config location')
+  }
+
+  let configData = null
+  if (isAbsoluteURL(configLocation)) {
```

Comment:
Could we also add a check here that the URL is only `http` or `https`? I think node-fetch won't accept the URL anyway if it isn't, but it might be nice to have an explicit error.

Reply:
Why should we care?  My thinking here was to let node-fetch return an error if the URL uses a scheme it can't handle.  If some future version of node-fetch supports ftp or gopher URLs, all the better :)

I'll double check that the error from node-fetch is actually clear enough though.

Reply:
So the node-fetch error actually isn't too bad:

> TypeError: Only HTTP(S) protocols are supported

It just lacks the context of which URL caused the failure.  We could certainly catch that and prepend that context...

``` javascript
try {
  fetch( ... )
} catch (e) {
  throw new Error(`Failed to fetch config from ${configLocation}: ${e}`)
}
```

That's roughly how you'd do it in Go, but I don't know if that's idiomatic in Node.  You likely lose any stacktrace info that was on the original error, but I don't know how much that matters in this case.

Reply:
Good point, the regular error is probably enough👍.